src/ls.c

```diff
@@ -3265,12 +3271,58 @@ cmp_extension (struct fileinfo const *a, struct fileinfo const *b,
   return diff ? diff : cmp (a->name, b->name);
 }
 
+/* Compare alphanumerically. That is, strings are sorted by their
+   alphanumeric components, with numeric components sorted numerically.
+   For example, "foo9" comes before "foo10".  
+   
+   Sort code from https://stackoverflow.com/a/1344056/338803  */
+
+static inline bool
+isdigit(char c)
+{ 
+  return '0' <= c && c <= '9';
+}
```

Comment:
There's probably a better function to use. I was trying to avoid adding any headers.